File: Sources/GlideComponents/Tokens/Colors.swift
import SwiftUI

@available(iOS 15.0, macOS 12.0, *)
public struct GlideColors {
    
    // MARK: - Gray Light Mode
    public struct GrayLight {
        public static let _25 = Color(red: 253/255, green: 253/255, blue: 253/255)
        public static let _50 = Color(red: 250/255, green: 250/255, blue: 250/255)
        public static let _100 = Color(red: 245/255, green: 245/255, blue: 245/255)
        public static let _200 = Color(red: 233/255, green: 234/255, blue: 235/255)
        public static let _300 = Color(red: 213/255, green: 215/255, blue: 218/255)
        public static let _400 = Color(red: 164/255, green: 167/255, blue: 174/255)
        public static let _500 = Color(red: 113/255, green: 118/255, blue: 128/255)
        public static let _600 = Color(red: 83/255, green: 88/255, blue: 98/255)
        public static let _700 = Color(red: 65/255, green: 70/255, blue: 81/255)
        public static let _800 = Color(red: 37/255, green: 43/255, blue: 55/255)
        public static let _900 = Color(red: 24/255, green: 29/255, blue: 39/255)
        public static let _950 = Color(red: 10/255, green: 13/255, blue: 18/255)
    }
    
    // MARK: - Gray Dark Mode
    public struct GrayDark {
        public static let _25 = Color(red: 250/255, green: 250/255, blue: 250/255)
        public static let _50 = Color(red: 247/255, green: 247/255, blue: 247/255)
        public static let _100 = Color(red: 240/255, green: 240/255, blue: 241/255)
        public static let _200 = Color(red: 236/255, green: 236/255, blue: 237/255)
        public static let _300 = Color(red: 206/255, green: 207/255, blue: 210/255)
        public static let _400 = Color(red: 148/255, green: 151/255, blue: 156/255)
        public static let _500 = Color(red: 133/255, green: 136/255, blue: 142/255)
        public static let _600 = Color(red: 97/255, green: 101/255, blue: 108/255)
        public static let _700 = Color(red: 55/255, green: 58/255, blue: 65/255)
        public static let _800 = Color(red: 34/255, green: 38/255, blue: 47/255)
        public static let _900 = Color(red: 19/255, green: 22/255, blue: 27/255)
        public static let _950 = Color(red: 12/255, green: 14/255, blue: 18/255)
    }
    
    // MARK: - Error Colors
    public struct Error {
        public static let _25 = Color(red: 255/255, green: 251/255, blue: 250/255)
        public static let _50 = Color(red: 254/255, green: 243/255, blue: 242/255)
        public static let _100 = Color(red: 254/255, green: 228/255, blue: 226/255)
        public static let _200 = Color(red: 254/255, green: 205/255, blue: 202/255)
        public static let _300 = Color(red: 253/255, green: 162/255, blue: 155/255)
        public static let _400 = Color(red: 249/255, green: 112/255, blue: 102/255)
        public static let _500 = Color(red: 240/255, green: 68/255, blue: 56/255)
        public static let _600 = Color(red: 217/255, green: 45/255, blue: 32/255)
        public static let _700 = Color(red: 180/255, green: 35/255, blue: 24/255)
        public static let _800 = Color(red: 145/255, green: 32/255, blue: 24/255)
        public static let _900 = Color(red: 122/255, green: 39/255, blue: 26/255)
        public static let _950 = Color(red: 85/255, green: 22/255, blue: 12/255)
    }
    
    // MARK: - Warning Colors
    public struct Warning {
        public static let _25 = Color(red: 255/255, green: 252/255, blue: 245/255)
        public static let _50 = Color(red: 255/255, green: 250/255, blue: 235/255)
        public static let _100 = Color(red: 254/255, green: 240/255, blue: 199/255)
        public static let _200 = Color(red: 254/255, green: 223/255, blue: 137/255)
        public static let _300 = Color(red: 254/255, green: 200/255, blue: 75/255)
        public static let _400 = Color(red: 253/255, green: 176/255, blue: 34/255)
        public static let _500 = Color(red: 247/255, green: 144/255, blue: 9/255)
        public static let _600 = Color(red: 220/255, green: 104/255, blue: 3/255)
        public static let _700 = Color(red: 181/255, green: 71/255, blue: 8/255)
        public static let _800 = Color(red: 147/255, green: 55/255, blue: 13/255)
        public static let _900 = Color(red: 122/255, green: 46/255, blue: 14/255)
        public static let _950 = Color(red: 78/255, green: 29/255, blue: 9/255)
    }
    
    // MARK: - Success Colors
    public struct Success {
        public static let _25 = Color(red: 246/255, green: 254/255, blue: 249/255)
        public static let _50 = Color(red: 236/255, green: 253/255, blue: 243/255)
        public static let _100 = Color(red: 220/255, green: 250/255, blue: 230/255)
        public static let _200 = Color(red: 171/255, green: 239/255, blue: 198/255)
        public static let _300 = Color(red: 117/255, green: 224/255, blue: 167/255)
        public static let _400 = Color(red: 71/255, green: 205/255, blue: 137/255)
        public static let _500 = Color(red: 23/255, green: 178/255, blue: 106/255)
        public static let _600 = Color(red: 7/255, green: 148/255, blue: 85/255)
        public static let _700 = Color(red: 6/255, green: 118/255, blue: 71/255)
        public static let _800 = Color(red: 8/255, green: 93/255, blue: 58/255)
        public static let _900 = Color(red: 7/255, green: 77/255, blue: 49/255)
        public static let _950 = Color(red: 5/255, green: 51/255, blue: 33/255)
    }
    
    // MARK: - Semantic Colors
    public static var primary: Color {
        return Success._600
    }
    
    public static var secondary: Color {
        return GrayLight._100
    }
    
    public static var destructive: Color {
        return Error._500
    }
    
    public static var background: Color {
        return Color(.systemBackground)
    }
    
    public static var surface: Color {
        return GrayLight._25
    }
    
    public static var textPrimary: Color {
        return GrayLight._900
    }
    
    public static var textSecondary: Color {
        return GrayLight._600
    }
    
    public static var border: Color {
        return GrayLight._200
    }
}

File: Storybook/src/tokens/colors.ts
import designTokens from '../design-tokens.json';

// Extract color tokens and convert to CSS custom properties
export const colorTokens = {
  // Gray Light Mode
  grayLight: {
    25: designTokens.color['gray (light mode)']['25'].value,
    50: designTokens.color['gray (light mode)']['50'].value,
    100: designTokens.color['gray (light mode)']['100'].value,
    200: designTokens.color['gray (light mode)']['200'].value,
    300: designTokens.color['gray (light mode)']['300'].value,
    400: designTokens.color['gray (light mode)']['400'].value,
    500: designTokens.color['gray (light mode)']['500'].value,
    600: designTokens.color['gray (light mode)']['600'].value,
    700: designTokens.color['gray (light mode)']['700'].value,
    800: designTokens.color['gray (light mode)']['800'].value,
    900: designTokens.color['gray (light mode)']['900'].value,
    950: designTokens.color['gray (light mode)']['950'].value,
  },
  
  // Gray Dark Mode
  grayDark: {
    25: designTokens.color['gray (dark mode)']['25'].value,
    50: designTokens.color['gray (dark mode)']['50'].value,
    100: designTokens.color['gray (dark mode)']['100'].value,
    200: designTokens.color['gray (dark mode)']['200'].value,
    300: designTokens.color['gray (dark mode)']['300'].value,
    400: designTokens.color['gray (dark mode)']['400'].value,
    500: designTokens.color['gray (dark mode)']['500'].value,
    600: designTokens.color['gray (dark mode)']['600'].value,
    700: designTokens.color['gray (dark mode)']['700'].value,
    800: designTokens.color['gray (dark mode)']['800'].value,
    900: designTokens.color['gray (dark mode)']['900'].value,
    950: designTokens.color['gray (dark mode)']['950'].value,
  },
  
  // Error
  error: {
    25: designTokens.color.error['25'].value,
    50: designTokens.color.error['50'].value,
    100: designTokens.color.error['100'].value,
    200: designTokens.color.error['200'].value,
    300: designTokens.color.error['300'].value,
    400: designTokens.color.error['400'].value,
    500: designTokens.color.error['500'].value,
    600: designTokens.color.error['600'].value,
    700: designTokens.color.error['700'].value,
    800: designTokens.color.error['800'].value,
    900: designTokens.color.error['900'].value,
    950: designTokens.color.error['950'].value,
  },
  
  // Warning
  warning: {
    25: designTokens.color.warning['25'].value,
    50: designTokens.color.warning['50'].value,
    100: designTokens.color.warning['100'].value,
    200: designTokens.color.warning['200'].value,
    300: designTokens.color.warning['300'].value,
    400: designTokens.color.warning['400'].value,
    500: designTokens.color.warning['500'].value,
    600: designTokens.color.warning['600'].value,
    700: designTokens.color.warning['700'].value,
    800: designTokens.color.warning['800'].value,
    900: designTokens.color.warning['900'].value,
    950: designTokens.color.warning['950'].value,
  },
  
  // Success
  success: {
    25: designTokens.color.success['25'].value,
    50: designTokens.color.success['50'].value,
    100: designTokens.color.success['100'].value,
    200: designTokens.color.success['200'].value,
    300: designTokens.color.success['300'].value,
    400: designTokens.color.success['400'].value,
    500: designTokens.color.success['500'].value,
    600: designTokens.color.success['600'].value,
    700: designTokens.color.success['700'].value,
    800: designTokens.color.success['800'].value,
    900: designTokens.color.success['900'].value,
    950: designTokens.color.success['950'].value,
  },
};

// Generate CSS custom properties
export const cssColorVariables = `
:root {
  /* Gray Light Mode */
  --color-gray-light-25: ${colorTokens.grayLight[25]};
  --color-gray-light-50: ${colorTokens.grayLight[50]};
  --color-gray-light-100: ${colorTokens.grayLight[100]};
  --color-gray-light-200: ${colorTokens.grayLight[200]};
  --color-gray-light-300: ${colorTokens.grayLight[300]};
  --color-gray-light-400: ${colorTokens.grayLight[400]};
  --color-gray-light-500: ${colorTokens.grayLight[500]};
  --color-gray-light-600: ${colorTokens.grayLight[600]};
  --color-gray-light-700: ${colorTokens.grayLight[700]};
  --color-gray-light-800: ${colorTokens.grayLight[800]};
  --color-gray-light-900: ${colorTokens.grayLight[900]};
  --color-gray-light-950: ${colorTokens.grayLight[950]};

  /* Gray Dark Mode */
  --color-gray-dark-25: ${colorTokens.grayDark[25]};
  --color-gray-dark-50: ${colorTokens.grayDark[50]};
  --color-gray-dark-100: ${colorTokens.grayDark[100]};
  --color-gray-dark-200: ${colorTokens.grayDark[200]};
  --color-gray-dark-300: ${colorTokens.grayDark[300]};
  --color-gray-dark-400: ${colorTokens.grayDark[400]};
  --color-gray-dark-500: ${colorTokens.grayDark[500]};
  --color-gray-dark-600: ${colorTokens.grayDark[600]};
  --color-gray-dark-700: ${colorTokens.grayDark[700]};
  --color-gray-dark-800: ${colorTokens.grayDark[800]};
  --color-gray-dark-900: ${colorTokens.grayDark[900]};
  --color-gray-dark-950: ${colorTokens.grayDark[950]};

  /* Error */
  --color-error-25: ${colorTokens.error[25]};
  --color-error-50: ${colorTokens.error[50]};
  --color-error-100: ${colorTokens.error[100]};
  --color-error-200: ${colorTokens.error[200]};
  --color-error-300: ${colorTokens.error[300]};
  --color-error-400: ${colorTokens.error[400]};
  --color-error-500: ${colorTokens.error[500]};
  --color-error-600: ${colorTokens.error[600]};
  --color-error-700: ${colorTokens.error[700]};
  --color-error-800: ${colorTokens.error[800]};
  --color-error-900: ${colorTokens.error[900]};
  --color-error-950: ${colorTokens.error[950]};

  /* Warning */
  --color-warning-25: ${colorTokens.warning[25]};
  --color-warning-50: ${colorTokens.warning[50]};
  --color-warning-100: ${colorTokens.warning[100]};
  --color-warning-200: ${colorTokens.warning[200]};
  --color-warning-300: ${colorTokens.warning[300]};
  --color-warning-400: ${colorTokens.warning[400]};
  --color-warning-500: ${colorTokens.warning[500]};
  --color-warning-600: ${colorTokens.warning[600]};
  --color-warning-700: ${colorTokens.warning[700]};
  --color-warning-800: ${colorTokens.warning[800]};
  --color-warning-900: ${colorTokens.warning[900]};
  --color-warning-950: ${colorTokens.warning[950]};

  /* Success */
  --color-success-25: ${colorTokens.success[25]};
  --color-success-50: ${colorTokens.success[50]};
  --color-success-100: ${colorTokens.success[100]};
  --color-success-200: ${colorTokens.success[200]};
  --color-success-300: ${colorTokens.success[300]};
  --color-success-400: ${colorTokens.success[400]};
  --color-success-500: ${colorTokens.success[500]};
  --color-success-600: ${colorTokens.success[600]};
  --color-success-700: ${colorTokens.success[700]};
  --color-success-800: ${colorTokens.success[800]};
  --color-success-900: ${colorTokens.success[900]};
  --color-success-950: ${colorTokens.success[950]};
}
`;